airbyte-webapp/src/hooks/services/ConfirmationModal/ConfirmationModalService.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useContext, useEffect, useMemo } from "react";
+
+import { ConfirmationModal } from "components/ConfirmationModal";
+
+import useTypesafeReducer from "hooks/useTypesafeReducer";
+
+import { ConfirmationModalOptions, ConfirmationModalServiceApi, ConfirmationModalState } from "./types";
+import { actions, initialState, confirmationModalServiceReducer } from "./reducer";
+
+const ConfirmationModalServiceContext = React.createContext<ConfirmationModalServiceApi | undefined>(undefined);
+
+export const useConfirmationModalService: (
+  confirmationModal?: ConfirmationModalOptions,
+  dependencies?: []
+) => {
+  openConfirmationModal: (confirmationModal: ConfirmationModalOptions) => void;
+  closeConfirmationModal: () => void;
+} = (confirmationModal, dependencies) => {
+  const confirmationModalService = useContext(ConfirmationModalServiceContext);
+  if (!confirmationModalService) {
+    throw new Error("useConfirmationModalService must be used within a ConfirmationModalService.");
+  }
+
+  useEffect(() => {
+    if (confirmationModal) {
+      confirmationModalService.openConfirmationModal(confirmationModal);
+    }
+    return () => {
+      if (confirmationModal) {
+        confirmationModalService.closeConfirmationModal();
+      }
+    };
+    // eslint-disable-next-line
+  }, [confirmationModal, confirmationModalService, ...(dependencies ?? [])]);
```

Comment:
Nitpick: You can set dependencies to `[]` in the params so this bit will look cleaner.